packages/context/src/inject.ts

```diff
@@ -331,19 +396,70 @@ export function describeInjectedArguments(
   return meta || [];
 }
 
-function resolveByTag(
+/**
+ * Inspect the target type
+ * @param injection
+ */
+function inspectTargetType(injection: Readonly<Injection>) {
+  let type = MetadataInspector.getDesignTypeForProperty(
+    injection.target,
+    injection.member!,
+  );
+  if (type) {
+    return type;
+  }
+  const designType = MetadataInspector.getDesignTypeForMethod(
+    injection.target,
+    injection.member!,
+  );
+  type =
+    designType.parameterTypes[
+      injection.methodDescriptorOrParameterIndex as number
+    ];
+  return type;
+}
+
+/**
+ * Resolve values by a binding filter function
+ * @param ctx Context object
+ * @param injection Injection information
+ * @param session Resolution session
+ */
+function resolveByFilter(
   ctx: Context,
   injection: Readonly<Injection>,
   session?: ResolutionSession,
 ) {
-  const tag: string | RegExp = injection.metadata!.tag;
-  const bindings = ctx.findByTag(tag);
+  const targetType = inspectTargetType(injection);
+  const decorator =
+    (injection.metadata && injection.metadata.decorator) || '@inject';
+  const targetName = ResolutionSession.describeInjection(injection)!.targetName;
+  if (decorator === '@inject.view') {
+    if (targetType && targetType !== ContextView) {
+      throw new Error(
+        `The type of ${targetName} (${targetType.name}) is not ContextView`,
+      );
+    }
+  } else if (decorator === '@inject') {
+    if (targetType !== Array) {
+      throw new Error(
+        `The type of ${targetName} (${targetType.name}) is not Array`,
+      );
+    }
+  }
+  const bindingFilter = injection.bindingSelector as BindingFilter;
+  const view = new ContextView(ctx, bindingFilter);
+  const autoOpen = injection.metadata!.autoOpen;
```

Comment:
What are the situations where the user will want to disable `autoOpen`? Can we leave this flag out of the initial implementation and wait until there is an actual demand for it?

Reply:
Removed `autoOpen`